tests/MyGame/Example/Stat.go

```diff
@@ -94,6 +94,34 @@ func (rcv *Stat) MutateCount(n uint16) bool {
 	return rcv._tab.MutateUint16Slot(8, n)
 }
 
+func StatKeyCompare(o1, o2 flatbuffers.UOffsetT, buf []byte) bool {
+	val1 := flatbuffers.GetUint16(buf[flatbuffers.GetFieldOffset(buf, 8, o1):])
+	val2 := flatbuffers.GetUint16(buf[flatbuffers.GetFieldOffset(buf, 8, o2):])
+	return val1 < val2
+}
+
+func (rcv *Stat) LookupByKey(key uint16, vectorLocation flatbuffers.UOffsetT, buf []byte) bool {
+	span := flatbuffers.GetUOffsetT(buf[vectorLocation - 4:])
+	start := flatbuffers.UOffsetT(0)
+	for span != 0 {
+		middle := span / 2
+		tableOffset := flatbuffers.GetIndirectOffset(buf, vectorLocation+ 4 * (start + middle))
```

Comment:
I know we had bugs in other languages to this part, where we try to read a field value directly from the buffer instead of using the public API getters. This matters since the getters include the default (not-written to the buffer) values that won't exist in buffer.

Reply:
See: https://github.com/google/flatbuffers/pull/7441

Reply:
Fixed. This cleans up alot of the code too.